packages/libs/src/services/solana/SolanaWeb3Manager.ts

```diff
@@ -702,23 +720,28 @@ export class SolanaWeb3Manager {
       data: Buffer.from(data)
     })
 
-    const instructions = [
-      transferInstruction,
-      paymentRouterInstruction,
-      memoInstruction
-    ]
-    return instructions
+    instructions.push(paymentRouterInstruction, memoInstruction)
+
+    const recentBlockhash = (await this.connection.getLatestBlockhash())
+      .blockhash
+    const transaction = new Transaction({
+      feePayer: this.feePayerKey,
+      recentBlockhash: recentBlockhash
+    }).add(...instructions)
+    if (!isSenderUserBank && senderKeypair) {
+      transaction.partialSign(senderKeypair)
```

Comment:
fwiw I don't think this part is necessary - the transaction we get back from coinflow is strips the signatures anyway. Maybe it's useful in other cases, but I think we can probably just have consumers sign the resulting transaction?

Reply:
this fn is actually not used at all i think, just wanted to include it in case we needed in the future.

Reply:
the fn is used, but this part is not right? the other two call this one? either way I think it's fine! just wanted to call it out since it was a learning for me wrt coinflow's relay

Reply:
sry my prev comment was erroneous (and i deleted it), ur totally right.
i'm still confused as to when the tx actually gets signed tho.